packages/opentelemetry-api/package.json

```diff
@@ -51,11 +51,13 @@
     "access": "public"
   },
   "dependencies": {
-    "@opentelemetry/context-base": "^0.14.0"
+    "@opentelemetry/context-base": "^0.14.0",
+    "semver": "^7.3.2"
```

Comment:
we don't need all functionality from semver and I would like to avoid using 3rd party dependency in api which is used for web too. For our purposes all we need is simple comparison, probably this is more than enough as we will always have version x.y.z and if x.y matches then it is enough. Having said that this is a function:

```
function satisfies(version1, version2) {
  return version1.substr(0, version1.lastIndexOf('.')) === version2.substr(0, version2.lastIndexOf('.'))
}
```

Reply:
or 
```
function satisfies(version1, version2) {
  const arr1 = version1.split('.');
}
```

Reply:
There are a lot of tricky edge cases to parsing and ordering semver. In the future we may need to have versions with `-alpha` tags and such for instance or build ids on nightlies. Ensuring the correct compatible version is chosen can be tricky business. What if we only import the `SemVer` class using `import * as SemVer from 'semver/classes/semver'`? This would make it much easier for tree shakers to only use the parts they need.

Reply:
[`bb718d9` (#1772)](https://github.com/open-telemetry/opentelemetry-js/pull/1772/commits/bb718d9260fa9318ed42bf15925be49dc26f1063)